src/groups/mqb/mqba/mqba_clientsession.cpp

```diff
@@ -3016,23 +3258,13 @@ void ClientSession::onLowWatermark()
 {
     // executed by the *IO* thread
 
-    if (!d_self.isValid()) {
```

Comment:
`d_self_sp` is always valid now because we call `ClientSession::onLowWatermark` when we acquire it